src/html5/parser/helper.rs

```diff
@@ -90,12 +90,7 @@ impl Html5Parser<'_> {
         }
     }
 
-    pub fn insert_text_helper(
-        &mut self,
-        node: NodeId,
-        position: InsertionPositionMode<NodeId>,
-        token: &Token,
-    ) {
+    pub fn insert_text_helper(&mut self, position: InsertionPositionMode<NodeId>, token: &Token) {
```

Comment:
When we merge texts, we don't need to create a node, so we are not passing one anymore and only create one when it's needed.